web/app/static/js/controllers/periscopeController.js

```diff
@@ -33,14 +33,25 @@ angular.module('uluwatuControllers').controller('periscopeController', ['$scope'
             }
         }, false);
 
+        $rootScope.$watch("activeCluster.cluster.status == 'AVAILABLE' && activeCluster.status == 'AVAILABLE'", function(newValue, oldValue) {
+            if ($rootScope.activeCluster.cluster != undefined && $rootScope.activeCluster.cluster.ambariServerIp != undefined) {
+                var periCluster = selectPeriscopeClusterByAmbariIp($rootScope.activeCluster.cluster.ambariServerIp);
+                if (newValue && !oldValue && isSelectedUluClusterEqualsPeriClusterAndRunning(periCluster)) {
+                    getAlertDefinitions(periCluster.id)
+                }
+            }
+        });
+
         function isSelectedUluClusterEqualsPeriClusterAndRunning(periCluster) {
             return periCluster != undefined && $rootScope.activeCluster.cluster.ambariServerIp != undefined && $rootScope.activeCluster.cluster.ambariServerIp == periCluster.host && periCluster.state == 'RUNNING';
         }
 
         function setActivePeriClusterWithResources(periscopeCluster) {
             $scope.actPeriscopeCluster = periscopeCluster;
             $scope.autoScalingSLAPoliciesEnabled = true;
-            getAlertDefinitions(periscopeCluster.id);
+            if ($rootScope.activeCluster.cluster.status == 'AVAILABLE' && $rootScope.activeCluster.status == 'AVAILABLE') {
+                getAlertDefinitions(periscopeCluster.id);
+            }
             getAlarms(periscopeCluster.id);
```

Comment:
If we do not show the div of autoscaling when the cluster and it's underlying stack is not available then we should not query other autoscaling related resources like scaling configs, scaling history, alarms, etc...